src/plugins/controls/server/options_list/options_list_queries.ts

```diff
@@ -139,6 +156,62 @@ const suggestionAggSubtypes: { [key: string]: OptionsListAggregationBuilder } =
       ),
   },
 
+  /**
+   * the "IP" query / parser should be used when the options list is built on a field of type IP.
+   */
+  ip: {
+    buildAggregation: ({ fieldName, searchString }: OptionsListRequestBody) => {
+      let rangeQuery: IpRangeQuery['rangeQuery'] = [
+        {
+          key: 'ipv6',
+          from: '::',
+          to: 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff',
+        },
+      ];
+      if (searchString) {
+        const { rangeQuery: testRangeQuery, validSearch } = getIpRangeQuery(searchString);
```

Comment:
Because the variable name `rangeQuery` was used above I just needed to give it a temporary name so I could destructure the entire object including `validSearch` - you're right it's confusing though 😆 

I modified the logic to hopefully be a bit more clear?

```typescript
let ipRangeQuery: IpRangeQuery = {
  validSearch: true,
  rangeQuery: [
    {
      ...
    },
  ],
};

if (searchString) {
  ipRangeQuery = getIpRangeQuery(searchString);
  ...
```